google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/util/AppEngineUtil.java

```diff
@@ -93,7 +93,8 @@ private AppEngineUtil() {
       Collection<Artifact> artifacts = ArtifactUtil.getArtifactsContainingModuleOutput(module);
       for (Artifact artifact : artifacts) {
         if ((isStandardModule && projectService.isAppEngineStandardArtifactType(artifact))
-            || (environment.isFlexible() && projectService.isAppEngineFlexArtifactType(artifact))) {
+            || (!isFlexCompat && environment.isFlexible()
```

Comment:
it's getting pretty difficult to parse the logic going on in this class.  We may want to consider a rethink/refactor sometime this quarter with some testing to help document the expectations here.